File: tests/benchmarks/resolution/fixtures/dynamic-kotlin/reflection.kt
// Fixture: Kotlin callable reference patterns
// ::greet → resolved as reflection kind
// invoke() → flagged as unresolved-dynamic

fun greet(name: String): String = "Hello, $name"

fun farewell(name: String): String = "Goodbye, $name"

// ::greet callable reference — resolved to top-level greet()
fun runCallableRef(): (String) -> String {
    return ::greet
}

// ::farewell callable reference — resolved to top-level farewell()
fun runFarewellRef(): (String) -> String {
    return ::farewell
}

// fn.invoke(args) — unresolved; fn could be any callable
fun runInvoke(fn: (String) -> String): String {
    return fn.invoke("world")
}

// Member callable reference: Greeter::greet must resolve to Greeter.greet,
// not the top-level greet() above, even though both share the same name.
class Greeter {
    fun greet(name: String): String = "Hi, $name"
}

fun runMemberCallableRef(): (Greeter, String) -> String {
    return Greeter::greet
}

File: tests/benchmarks/resolution/fixtures/dynamic-csharp/Reflection.cs
// Phase 6: C# reflection patterns
// GetMethod("name") → reflection kind
// method.Invoke() → unresolved-dynamic
using System.Reflection;

public class Reflection {
    public static string Greet(string name) => $"Hello, {name}";

    // type.GetMethod("Greet") — reflection kind, resolvable in theory
    public static MethodInfo RunGetMethod(Type type) {
        return type.GetMethod("Greet");
    }

    // method.Invoke(target, args) — unresolved-dynamic
    public static object RunInvoke(MethodInfo method, object target) {
        return method.Invoke(target, new object[] { "world" });
    }
}
